staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1322,14 +1326,17 @@ func (c *Cacher) getBookmarkAfterResourceVersionLockedFunc(parsedResourceVersion
 //
 //	if SendInitiaEvents != nil => ResourceVersionMatch = NotOlderThan
 //	if ResourceVersionmatch != nil => ResourceVersionMatch = NotOlderThan & SendInitialEvents != nil
-func (c *Cacher) getWatchCacheResourceVersion(ctx context.Context, parsedWatchResourceVersion uint64, opts storage.ListOptions) (uint64, error) {
+func (c *Cacher) getWatchCacheResourceVersion(ctx context.Context, parsedWatchResourceVersion uint64, opts storage.ListOptions, isRequestWatchProgressSupported bool) (uint64, error) {
 	if len(opts.ResourceVersion) != 0 {
 		return parsedWatchResourceVersion, nil
 	}
 	// legacy case
 	if !utilfeature.DefaultFeatureGate.Enabled(features.WatchFromStorageWithoutResourceVersion) && opts.SendInitialEvents == nil && opts.ResourceVersion == "" {
 		return 0, nil
 	}
+	if isListWatchRequest(opts) && !isRequestWatchProgressSupported {
```

Comment:
I'm not sure that's the right place - you can have watch-list request also with a given RV (line 1330), and we probably want to fail for such case too.

I think that this check should be moved out of this function and be called directly before `getWatchCacheResourceVersion` function is called.

Reply:
ok, good point, thanks.